packages/js/src/logger.ts

```diff
@@ -0,0 +1,95 @@
+import { Axiom } from "./client";
+import { LogEvent, LogLevel, LoggerOptions } from "./type";
+import { jsonFriendlyErrorReplacer, prettyPrint, throttle } from "./utils";
+
+class Logger {
+    public dataset: string;
+    private logLevel: LogLevel;
+    private LOG_LEVEL = process.env.AXIOM_LOG_LEVEL || 'debug';
+    public logEvents: LogEvent[] = [];
+    children: Axiom[] = [];
+    throttledSendLogs = throttle(this.sendLogs.bind(this), 1000);
+
+    constructor(public client: Axiom, public options: LoggerOptions) {
+        this.dataset = options.dataset;
+        this.logLevel = this.calculateLogLevel(options.logLevel);
+    }
+
+    debug = (message: string, args: { [key: string]: any } = {}) => {
+        this._log(LogLevel.debug, message, args);
+    };
+
+    info = (message: string, args: { [key: string]: any } = {}) => {
+        this._log(LogLevel.info, message, args);
+    };
+
+    warn = (message: string, args: { [key: string]: any } = {}) => {
+        this._log(LogLevel.warn, message, args);
+    };
+
+    error = (message: string, args: { [key: string]: any } = {}) => {
+        this._log(LogLevel.error, message, args);
+    };
+
+    private _log = (level: LogLevel, message: string, args: { [key: string]: any } = {}) => {
+        if (level < this.logLevel) {
+            return;
+        }
+        const logEvent: LogEvent = {
+            level: LogLevel[level].toString(),
+            message,
+            _time: new Date(Date.now()).toISOString(),
+            fields: this.options.args || {},
+        };
+        if (args instanceof Error) {
+            logEvent.fields = { ...logEvent.fields, message: args.message, stack: args.stack, name: args.name };
+        } else if (typeof args === 'object' && args !== null && Object.keys(args).length > 0) {
+            const parsedArgs = JSON.parse(JSON.stringify(args, jsonFriendlyErrorReplacer));
+            logEvent.fields = { ...logEvent.fields, ...parsedArgs };
+        } else if (args && args.length) {
+            logEvent.fields = { ...logEvent.fields, args: args };
+        }
+        this.logEvents.push(logEvent);
+        if (this.options.autoFlush) {
+            this.throttledSendLogs();
+        }
+    };
+
+    private sendLogs() {
+        console.log("sending logs")
+        if (!this.logEvents.length) {
+            return;
+        }
+        const dataset = this.options.dataset;
+
+        if (!dataset) {
```

Comment:
One of the things that were asked in next-axiom is that users want to control when the logger should sent to Axiom, and when to just print locally, So we introduced an env var for that, maybe we could keep it. I tried to look for it in next-axiom, but couldn't find it, my memory betrayed me, I guess when env vars are missing we just print to console. 

The only thing I found is this var that controls pretty printing.

```
process.env.AXIOM_NO_PRETTY_PRINT
```